Vu Instruction Implementations

astrelsky opened the pull request:
Still need to implement the pcodeInjector but this should cover most of the instructions. Also the context register needs to be set manually at the moment. Running the small script below when the currentAddress is within .vutext or .vudata should do the trick for now.
```
#@category ghidra-emotionengine

from java.math import BigInteger

if __name__ == '__main__':
    block = getMemoryBlock(currentAddress)
    pc = currentProgram.getProgramContext()
    uM = pc.getRegister('microMode')
    pc.setValue(uM, block.start, block.end, BigInteger.ONE)
    #disassemble(currentAddress)
```

chaoticgd commented:
Some preliminary discussion here: https://github.com/chaoticgd/ghidra-emotionengine/pull/2

Commit:
Merge branch 'vu' of https://github.com/chaoticgd/ghidra-emotionengine

Commit:
Added ghidra libs, sources and documentation...

to the classpath so they can be found by eclipse jdtls.
This is for working with ghidra java projects in vscode.

Commit:
VU instruction implementations

astrelsky commented:
Ok, I have updated the branch in the pr. Everything should decompile well with the exception of a few quirks with matrices due to a bug in ghidra [bug link](https://github.com/NationalSecurityAgency/ghidra/issues/1180).

I'm attaching the decompiled results from decompiling [ps2dev/ps2sdk/libvux/vuhw.o](https://github.com/ps2dev/ps2sdk/blob/master/ee/libvux/src/vuhw.c). I removed unnecessary definitions and types and corrected the vector and matrix typedefs from the output as well as attaching the ps2dev copyright header since it is technically a derived work.

Also it will take a minute or two for sleigh to compile the files the first time the processor is loaded. This gets better in ghidra 9.2.

[vuhw.tar.gz](https://github.com/beardypig/ghidra-emotionengine/files/3828708/vuhw.tar.gz)

beardypig commented:
This includes all the changes from #23 right?

astrelsky commented:
Yes.

beardypig commented:
Sorry @astrelsky, there were some conflicts, could you resolve them? Or is it better to revert #23 and just use this PR?

astrelsky commented:
I can resolve them in few minutes.

Commit:
Merge branch 'vu' of https://github.com/astrelsky/ghidra-emotionengine into vu
